synapse/_scripts/synapse_port_db.py

```diff
@@ -1093,7 +1105,7 @@ async def _setup_sequence(
                 await self.sqlite_store.db_pool.simple_select_one_onecol(
                     table=stream_id_table,
                     keyvalues={},
-                    retcol="COALESCE(MAX(stream_id), 1)",
+                    retcol=f"COALESCE(MAX({column_name}), 1)",
```

Comment:
I think this is fine as-is, but I have some thoughts as an aside.

The error message quoted in the logs of #16040 uses `SELECT GREATEST(MAX(txn_id), 0)`. Is there an important difference between 0 and 1 as a fallback here? I don't _think_ so...

I also vaguely remember reading some stream code that suggests streams could count backwards.

https://github.com/matrix-org/synapse/blob/02f74f3a997a4356b5bda957ebc51a829dad15f9/synapse/storage/util/id_generators.py#L180-L181

If negative-step streams are used gets used, I think this logic will be broken. But I suspect we never use it (and would vote to remove it).

Reply:
I think we at some point changed these to start at `1` instead of `0` in some situations? I don't think there's much of a difference honestly. I didn't understand the fixes there though, see #13766.

(Note that this coalesces to `1`, but then adds `1` below so it is either 0 or 2.)